### PR TITLE
Honor minio fullname

### LINKED ISSUE #9564: fullnameOverride mixing in helm subcharts
#### Describe the bug
If you set the fullnameOverride for mimir and for subcharts, some manifests are generated bad. 

#### To Reproduce
Sample values.yaml:

```
---
fullnameOverride: grafana-mimir
minio:
  fullnameOverride: grafana-mimir-minio

```
Generate the manifest: 
`helm template --dry-run --debug -f values.yaml --repo https://grafana.github.io/helm-charts mimir-distributed --name-template csiga --output-dir out`

Check the main templates, its good:
```
$ grep " name:" out/mimir-distributed/templates/runtime-configmap.yaml 
  name: grafana-mimir-runtime
```
Check a minio template, its good:
```
$ grep " name:" out/mimir-distributed/charts/minio/templates/configmap.yaml 
  name: grafana-mimir-minio

```
But some template have bad reference, like:
```
$ grep -A1 configMap out/mimir-distributed/templates/minio/create-bucket-job.yaml 
            - configMap:
                name: csiga-minio
```

#### Expected behavior

create-bucket-job.yaml sould have configmap name: grafana-mimir-minio

#### Environment

- Infrastructure: kubernetes
- Deployment tool: helm

### PR DESCRIPTION
**Pull request by elsoa-invitech**
#### What this PR does

the subchart full name override is not honored is referenced template files. (eg minio service name or configmap name)

#### Which issue(s) this PR fixes or relates to

Fixes #9564.

### CODE REVIEW

**Comment by elsoa-invitech** [comment]
Solution from here: https://stackoverflow.com/a/69515809

File: operations/helm/charts/mimir-distributed/templates/minio/create-bucket-job.yaml
Comment (narqo) [inline operations/helm/charts/mimir-distributed/templates/minio/create-bucket-job.yaml:1]: Could you also fix the value passed to `MINIO_ENDPOINT` in this template. It still refers to the wrong svc name.

**Comment by narqo** [comment]
The fix looks good to me. But I think there is one place in the changeset, that also needs fixing (refer to the in place comment).

Also, could you add a `BUGFIX` entry to the chart's CHANGELOG in the `operations/helm/charts/mimir-distributed/CHANGELOG.md` file.

**Comment by elsoa-invitech** [comment]
anyway, what do you think: in values.yaml mimir config endpoint sould use `{{ .Values.minio.service.port }}` instead of 9000?

**Comment by narqo** [comment]
> in `values.yaml` mimir config endpoint sould use `{{ .Values.minio.service.port }}` instead of 9000?

That's a great point, thank you. Fixing it should be a noop for the chart and its tests so if you could fix the values here, that'd be nice.

**Comment by elsoa-invitech** [comment]
Changed the port to minio.service values

**Review by narqo — APPROVED** [review]
🔥 Looks good, thank you

**Pull request merged**